mpl_data_cast/gui/main.py

```diff
@@ -0,0 +1,235 @@
+import traceback
+import pathlib
+import dclab
+import h5py
+import numpy
+import pkg_resources
+from PyQt6 import uic, QtCore, QtGui, QtWidgets
+
+from . import preferences
+from ..mod_recipes.rcp_rtdc import RTDCRecipe
+from .._version import version
+
+# set Qt icon theme search path
+QtGui.QIcon.setThemeSearchPaths([
+    os.path.join(pkg_resources.resource_filename("mpl_data_cast", "img"),
+                 "icon-theme")])
+QtGui.QIcon.setThemeName(".")
+
+
+class MPLDataCast(QtWidgets.QMainWindow):
+    def __init__(self, *args, **kwargs):
+        """Initialize MPL-Data-Cast"""
+        QtCore.QCoreApplication.setOrganizationName("GuckLab")
+        QtCore.QCoreApplication.setApplicationName("MPL-Data-Cast")
+        super(MPLDataCast, self).__init__(*args, **kwargs)
+
+        QtWidgets.QMainWindow.__init__(self)
+        path_ui = pkg_resources.resource_filename("mpl_data_cast.gui",
+                                                  "main.ui")
+        uic.loadUi(path_ui, self)
+
+        # settings
+        QtCore.QSettings.setDefaultFormat(QtCore.QSettings.Format.IniFormat)
+        self.settings = QtCore.QSettings()
+        # self.settings.setIniCodec("utf-8")
+
+        # load some values from the settings
+        self.widget_output.tree_depth_limit = int(self.settings.value(
+            "rtdc/tree_depth_limit", 3))
+        self.widget_input.tree_depth_limit = int(self.settings.value(
+            "rtdc/tree_depth_limit", 3))
+        self.widget_output.update_output_dir(
+            self.settings.value("rtdc/output_path", str(pathlib.Path.cwd())))
+        # signals
+        self.pushButton_transfer.clicked.connect(self.on_task_transfer)
+        # GUI
+        self.setWindowTitle(f"MPL-Data-Cast {version}")
+        # Disable native menu bar (e.g. on Mac)
+        self.menubar.setNativeMenuBar(False)
+        # File menu
+        self.actionPreferences.triggered.connect(self.on_action_preferences)
+        self.actionQuit.triggered.connect(self.on_action_quit)
+        # Help menu
+        self.actionSoftware.triggered.connect(self.on_action_software)
+        self.actionAbout.triggered.connect(self.on_action_about)
+
+        self.show()
+        self.raise_()
+
+    @QtCore.pyqtSlot()
+    def on_action_preferences(self):
+        """Show the preferences dialog"""
+        prev_tree_depth_limit = self.settings.value("rtdc/tree_depth_limit", 8)
+        dlg = preferences.Preferences(self)
+        dlg.setWindowTitle("MPL-Data-Cast Preferences")
+        dlg.exec()
+        # update maximum tree depth if necessary
+        if self.settings.value("rtdc/tree_depth_limit",
+                               8) != prev_tree_depth_limit:
+            self.widget_output.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            self.widget_input.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            if self.widget_output.path is not None:
```

Comment:
Is this indentation level correct for this if-statement?

Reply:
Yes, the visualization of the files and folders needs to be updated only if the tree depth limit has changed.

Reply:
OK thanks, could you please add this as a comment here?